### DISCUSSION TITLE
Rationale and methodology of adding non-LCG scripts?

### DISCUSSION THREAD

**Discussion by jamadagni** [microsoft/cascadia-code#780] [General]
Hello. As far as I can tell LCG (Latin, Cyrillic and Greek) scripts share many common factors (due to geography) so it is meaning to include all of them in one font or family, especially monospace.

However I note that Cascadia also includes support for Middle East scripts Arabic and Hebrew. Recently also support for Far East scripts Chinese and Japanese has been added.

From thread #762 it seems that the *rationale* for adding Far East is to ensure glyphs from those scripts can also fit within the grid and the *methodology* is that they will have twice the width though I didn't see this explicitly mentioned anywhere. I am not sure if this is to be implicitly understood?

I would also like to know the rationale and methodology for Middle East glyphs as well.

The reason I ask this is I would like to know if there is meaning in also including Indic scripts like Devanagari and Tamil in monospace fonts. I use these scripts as part of my programming requirements. 

Here's a app/lib I wrote in Python with *all* symbols from my side being in Tamil: [Yaappu](https://github.com/jamadagni/yaappu).

**Comment by aaronbell** [comment]
> rationale for adding Far East is to ensure glyphs from those scripts can also fit within the grid 

The rationale is that we have noted a lot of interest in coding fonts that include CJK characters—I've seen a number of cases where Cascadia has been merged with other CJK fonts to create a font that is optimized for that use case. 

> methodology is that they will have twice the width though I didn't see this explicitly mentioned anywhere. I am not sure if this is to be implicitly understood?

Historically, CJK glyphs are categorized by coding environments in Windows as being double-width. Thus, it is important to have a font that is compatible with that requirement. 

For middle eastern glyphs, Windows Terminal was implementing RTL support and wanted to be able to demonstrate it—thus Arabic and Hebrew. 

I certainly have an interest in including Devanagari and Tamil, but there needs to be sufficient support to justify the cost of the work being commissioned. I'd recommend trying to get more people to upvote this discussion, or #151 to demonstrate to Microsoft a desire for coding langauge support for Indic scripts.
